include/misc/mempool.h

```diff
@@ -0,0 +1,97 @@
+#ifndef SYS_MEMPOOL_H
+#define SYS_MEMPOOL_H
+
+#include <kernel.h>
+#include <misc/mempool_base.h>
+
+/**
+ * @brief Statically define and initialize a memory pool.
+ *
+ * The memory pool's buffer contains @a n_max blocks that are @a max_size bytes
+ * long. The memory pool allows blocks to be repeatedly partitioned into
+ * quarters, down to blocks of @a min_size bytes long. The buffer is aligned
+ * to a @a align -byte boundary.
+ *
+ * If the pool is to be accessed outside the module where it is defined, it
+ * can be declared via
+ *
+ * @code extern struct sys_mem_pool <name>; @endcode
+ *
+ * @param name Name of the memory pool.
+ * @param minsz Size of the smallest blocks in the pool (in bytes).
+ * @param maxsz Size of the largest blocks in the pool (in bytes).
+ * @param nmax Number of maximum sized blocks in the pool.
+ * @param align Alignment of the pool's buffer (power of 2).
+ */
+#define SYS_MEM_POOL_DEFINE(name, minsz, maxsz, nmax, align)		\
+	char __aligned(align) _mpool_buf_##name[_ALIGN4(maxsz * nmax)	\
+				  + _MPOOL_BITS_SIZE(maxsz, minsz, nmax)]; \
+	struct sys_mem_pool_lvl _mpool_lvls_##name[_MPOOL_LVLS(maxsz, minsz)]; \
```

Comment:
I wonder how this is going to work when we have ASLR.

Reply:
I'm deleting this macro.
We're gonna set these up at runtime.